.github/workflows/haskell.yml

```diff
@@ -31,24 +32,71 @@ jobs:
       with:
         ghc-version: ${{ matrix.ghc }}
         cabal-version: ${{ matrix.cabal }}
+        enable-stack: ${{ matrix.use-stack }}
+
+    - name: Set stack environment variable
+      if: matrix.use-stack
+      run: echo '::set-env name=USE_STACK::TRUE'
 
     - name: Cache
-      uses: actions/cache@v1
+      uses: actions/cache@v2
       env:
-        cache-name: cache-cabal
+        cache-name: build-cache
       with:
-        path: ~/.cabal
-        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/*.cabal') }}-${{ hashFiles('**/cabal.project') }}
+        path: |
+            ~/.cabal
+            ~/.stack
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/*.cabal') }}-${{ hashFiles('**/stack.yaml') }}
         restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/*.cabal') }}-${{ hashFiles('**/stack.yaml') }}
           ${{ runner.os }}-build-${{ env.cache-name }}-
           ${{ runner.os }}-build-
           ${{ runner.os }}-
 
-    - name: Install dependencies
+    - name: Before install (Linux)
+      if: matrix.os == 'ubuntu-latest'
+      run: |
+        mkdir -p /tmp/minio /tmp/minio-config/certs
+        cp test/cert/* /tmp/minio-config/certs/
+        (cd /tmp/minio; wget https://dl.min.io/server/minio/release/linux-amd64/minio; chmod +x ./minio)
+        sudo cp /tmp/certs/public.crt /usr/local/share/ca-certificates/
+        sudo update-ca-certificates
+        MINIO_ACCESS_KEY=minio MINIO_SECRET_KEY=minio123 /tmp/minio/minio server --quiet --certs-dir /tmp/certs data 2>&1 > minio.log &
```

Comment:
enable multi-disk so that we can eventually test versioning as well @donatello ?

Reply:
Sure, if I get this basic thing working, I'll get to that.